plugins/modules/gpg_keypair.py

```diff
@@ -0,0 +1,507 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2024, Austin Lucas Lake <53884490+austinlucaslake@users.noreply.github.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: gpg_keypair
+author: "Austin Lucas Lake (@austinlucaslake)"
+short_description: Generate or delete GPG private and public keys
+version_added: 2.20.0
+description:
+    - "This module allows one to generate or delete GPG private and public keys using GnuPG (gpg)."
+requirements:
+    - gpg >= 2.1
```

Comment:
You need the default `attributes` docs fragment:
```suggestion
    - gpg >= 2.1
extends_documentation_fragment:
  - community.crypto.attributes
```